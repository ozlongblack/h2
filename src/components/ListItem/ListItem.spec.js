// @flow

import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';
import type { ListItemProps } from './ListItem.props';
import ListItemView from './ListItem.view';

describe('ListItem', () => {
  const RealDate: * = Date;

  function mockDate (isoDate: string) {
    global.Date = class extends RealDate {
      constructor (): Date {
        return new RealDate(isoDate)
      }
    }
  }

  afterEach(() => {
    global.Date = RealDate;
  });

  it('renders correctly', () => {
    mockDate('2017-03-25T12:30:00z');

    const fixture: ListItemProps = {
      category: 'category',
      description: 'description',
      intl: {
        formatDate: jest.fn(),
      }
    };

    const tree: JSON = renderer
      .create(
        <IntlProvider locale="en">
          <ListItemView {...fixture} />
        </IntlProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

