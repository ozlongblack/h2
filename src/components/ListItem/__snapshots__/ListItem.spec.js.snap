// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
<div>
  <div>
    <span>
      3/25/2017
    </span>
    <span>
      category
    </span>
  </div>
  <div>
    description
  </div>
</div>
`;
